src/views/shared/SaunaDetailBreadcrumb.tsx

```diff
@@ -0,0 +1,30 @@
+import { ReactElement, useEffect, useState } from 'react'
+import { BreadcrumbComponentProps } from 'use-react-router-breadcrumbs'
+import { Sauna } from '../../entities/Sauna'
+import api from '../../networking/api'
+
+/**
+ * Dynamic breadcrumb which display the name of the sauna in the
+ * navigation stack. To get the name, it is fetched by id from the
+ * API.
+ */
+function SaunaDetailBreadcrumb({ match, testId }: SaunaDetailBreadcrumbProps<string>): ReactElement {
+    const saunaId = Number(match.params['saunaId'])
+    const [sauna, setSauna] = useState<Sauna.Request>(Sauna.emptyRequest())
```

Comment:
I would use `Sauna.Response` here with an undefined initial value. You could set the default value directly in the span below like `sauna?.name || 'Details'`.

Reply:
Nice